### ISSUE TITLE
About grad_accumulation_steps

### ISSUE THREAD

**Issue by RayYoh** [saidwivedi/InteractVLM#5]
Hi authors, thanks for your work. I see most of the training code is based on LISA, but I find there are some differences. 
For gradient accumulation, I see your code uses if-else to judge whether to update the parameters ([here](https://github.com/saidwivedi/InteractVLM/blob/88d12aeaefd4b7b22b2b821ca1029ce19789008e/train.py#L552)), however, in LISA, they didn't use this, and the author explained that deepseed can process this [issue](https://github.com/dvlab-research/LISA/issues/58#issuecomment-1735528256).

**Comment by saidwivedi** [comment]
@RayYoh thank you for your interest!

Great observation! Yes, we manually control the gradient accumulation to skip batches where the loss is NaN, which can occasionally occur during the early training iterations.. In our experiments, we use gradient accumulation as 1. However, in case of gradient accumulation more than 1, the `gradient_accumulation_steps` in `ds_config` (see [here](https://github.com/saidwivedi/InteractVLM/blob/88d12aeaefd4b7b22b2b821ca1029ce19789008e/train.py#L352)) should be set to 1 to avoid double accumulation since accumulation is already handled manually in the training loop.

**Comment by RayYoh** [comment]
Got that. Thanks for your patient reply!
